components/lock.rst

```diff
@@ -68,6 +68,14 @@ method can be safely called repeatedly, even if the lock is already acquired.
     on instance destruction. In some cases, it can be useful to lock a resource
     across several requests. To disable the automatic release behavior, set the
     third argument of the ``createLock()`` method to ``false``.
+    
+.. caution::
+
+    A common pitfall might be to use the ``isAcquired()`` method to check if
+    a lock has already been acquired before. However, this method is used to check
+    if the lock has been acquired by the **current process** and thus will not work
+    as expected across multiple processes.
```

Comment:
What's about describing the purpose of the method `isAquired` and how checking if the a lock is already acquired before?


here 2 sample. Don't hesitate to reword.

> because of race condition, the best way to check if a token is acquired by someone else, is to try to acquire it.
```
if ($lock->acquire()) {
  // if we are here, it means that nobody else had the lock. And nobodyelse would be able to get it until releasing it.
  $lock->release();
}
```

> the method `isAcquired` allow you to check if you are still the owner of the lock when performing a sensitive task.

```
if (!$lock->acquire()) {
  return;
}
$this->beginTransaction();

// perform a long process

if ($lock->isAcquired()) {
  $this->commit();
} else {
  $this->rollback();
  throw new \Exception('Process failed');
}
```

Reply:
Okay, so I'm trying to document this but I still don't quite get it and without getting it myself, I cannot document it 😄 So maybe we need a more tangible example here. So when I use a lock service then I want to make sure, nobody else can do anything until I'm done. This is the `if ($lock->acquire()) { // do stuff $lock->release(); }` example. That seems fine to me. But I don't understand when exactly I would use `->isAcquired()`. I mean, nobody can just take over my lock right? So why would I need to check if I'm the owner? I just can't come up with any use case.

Reply:
Mhh, I didn't mention that some lock store (Redis, Memcached, Pdo..) have a TTL (by design the store can not be aware when a client die (ie. segfault) that why they automatically release the lock after a defined amount of time. see https://symfony.com/doc/current/components/lock.html#expiring-locks). Thats why you technically can acquire a lock and lost it.

Reply:
Ah, of course! Thanks for the hint! Good, I know what I'm going to write now and I'll ping you for another review when I think I'm done  😄